### PR TITLE
[spec] Fix a bunch of minor issues

### PR DESCRIPTION
**Pull request by rossberg**
Discovered while working on WebAssembly/function-references#49.

### CODE REVIEW

File: document/core/binary/types.rst [hunk 1]
```diff
@@ -6,8 +6,8 @@ Types
 -----
 
 .. note::
-   In future versions of WebAssembly, value types may include types denoted by :ref:`type indices <syntax-typeidx>`.
-   Thus, the binary format for types corresponds to the encodings of small negative :math:`\xref{binary/values}{binary-sint}{\sN}` values, so that they can coexist with (positive) type indices in the future.
+   In some places, possible types include both type constructors or types denoted by :ref:`type indices <syntax-typeidx>`.
+   Thus, the binary format for most type constructors corresponds to the encodings of small negative :math:`\xref{binary/values}{binary-sint}{\sN}` values, such that they can unambiguously occur in the same place as (positive) type indices.
```
Comment (conrad-watt) [inline document/core/binary/types.rst:10]: "most type constructors" - is it possible for a type constructor to be represented by something other than a negative number?

**Review by conrad-watt — COMMENTED** [review]
Changes LGTM

some other references to "bit width" should be changed from syntax-valtype to syntax-numtype

one mention here (memory instructions):
https://webassembly.github.io/spec/core/syntax/instructions.html?highlight=bit%20width#memory-instructions

two mentions here (load and store)
https://webassembly.github.io/spec/core/exec/instructions.html?highlight=bit%20width#memory-instructions

**Review by rossberg — COMMENTED** [review]
Ah, thanks. Fixed, plus a couple more places where it should now be numtype. PTAL.

**Review by conrad-watt — APPROVED** [review]
LGTM

**Pull request merged**